backend/business-partner-agent/src/main/java/org/hyperledger/bpa/impl/aries/ProofManager.java

```diff
@@ -154,6 +240,42 @@ public void handleProofEvent(PresentationExchangeRecord proof) {
                         }));
     }
 
+    // handles all proof request
+    public void handleProofRequestEvent(PresentationExchangeRecord proof) {
+        partnerRepo.findByConnectionId(proof.getConnectionId()).ifPresent(
+                p -> {
+                    pProofRepo.findByPresentationExchangeId(proof.getPresentationExchangeId())
+                            .ifPresentOrElse(pProof -> {
+                                if (PresentationExchangeState.PROPOSAL_SENT.equals(pProof.getState()) &&
+                                        PresentationExchangeState.REQUEST_RECEIVED.equals(proof.getState()) &&
+                                        PresentationExchangeInitiator.SELF.equals(proof.getInitiator())) {
+                                    log.info(
+                                            "Present_Proof: state=request_received on PresentationExchange where initator=self, responding immediatly");
+                                    presentProof(proof);
+                                }
+
+                            }, () -> {
+                                if (PresentationExchangeRole.PROVER.equals(proof.getRole())
```

Comment:
Already checked in the event handler

Reply:
Ok, I  have seen you response. This is getting messy so we need to get this PR ready and then I will break this all up again.

Reply:
same as i previously mentioned. that's true for now, but this is a public method so that's no guaranteed. We should probably make an entire class to handle 'present_proof' events.